ui/src/containers/GroupFolderInput.jsx

```diff
@@ -1,81 +1,56 @@
-import React from 'react';
+import { useEffect } from 'react';
 import { Button, Form } from 'react-bootstrap';
-import { connect } from 'react-redux';
-import { bindActionCreators } from 'redux';
+import { useForm } from 'react-hook-form';
+import { useDispatch, useSelector } from 'react-redux';
 
 import { setGroupFolder } from '../actions/queue';
 
-class GroupFolderInput extends React.Component {
-  constructor(props) {
-    super(props);
-    this.inputOnChangeHandler = this.inputOnChangeHandler.bind(this);
-    this.setGroupFolderInput = this.setGroupFolderInput.bind(this);
-    this.inputValue = '';
-    this.state = { validationState: 'success' };
+function GroupFolderInput() {
+  const dispatch = useDispatch();
+  const groupFolder = useSelector((state) => state.queue.groupFolder);
+
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { isDirty },
+  } = useForm({
+    defaultValues: {
+      groupFolder: groupFolder || '',
+    },
+  });
+
+  useEffect(() => {
+    reset({ groupFolder: groupFolder || '' });
+  }, [groupFolder, reset]);
+
+  function onSubmit(data) {
+    dispatch(setGroupFolder(data.groupFolder));
   }
 
-  setGroupFolderInput() {
-    this.setState({ validationState: 'success' });
-    this.props.setGroupFolder(this.inputValue.value);
-  }
-
-  inputOnSelectHandler(e) {
-    e.stopPropagation();
-    e.nativeEvent.stopImmediatePropagation();
-  }
-
-  inputOnChangeHandler() {
-    this.setState({ validationState: 'warning' });
-  }
-
-  render() {
-    return (
-      <span>
-        <Form.Label>Group path :</Form.Label>
-        <br />
-        <Form
-          onSubmit={(e) => {
-            e.preventDefault();
+  return (
+    <Form as="div">
+      <Form.Label>Group path :</Form.Label>
```

Comment:
```suggestion
      <Form.Label>Group path</Form.Label>
```

Might as well :shrug: